### PR TITLE
DOC: initial tags for statistics section of gallery

### PR DESCRIPTION
**Pull request by story645**
## PR summary

Batch of tags for the stats functions, started by @ktnharris22 in #27235

I mostly didn't add to the tags except to flag stuff that's veering into tutorial or might belong in plot-types.  I put/propose stashing internal-tags in an ifconfig b/c I think they can be useful on dev as a flag of what to work on and confusing/clutter on any of the stable versions.  

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
Cycling to rebuild docs, now that upstream has been updated

File: galleries/examples/statistics/histogram_cumulative.py
Comment (QuLogic) [inline galleries/examples/statistics/histogram_cumulative.py:70]:
```suggestion
# .. tags:: plot-type: ecdf, plot-type: histogram, domain: statistics
```

File: galleries/examples/statistics/multiple_histograms_side_by_side.py
Comment (QuLogic) [inline galleries/examples/statistics/multiple_histograms_side_by_side.py:66]:
```suggestion
# .. tags:: domain: statistics, plot-type: barh, plot-type: histogram, styling: position
```

File: galleries/examples/statistics/violinplot.py
Comment (QuLogic) [inline galleries/examples/statistics/violinplot.py:109]:
```suggestion
# .. tags:: plot-type: violin, domain: statistics
```

**Review by QuLogic — APPROVED** [review]
This also needs a rebase.

**Comment by QuLogic** [comment]
The build is broken because of #28820, but fixing that, it's _really_ broken because `internal: too-much` is not a valid tag.

**Comment by timhoffm** [comment]
Well, I feel uncomfortable, with this duct-tape coding of internal tags via `.. ifconfig`. It's cluttered and difficult to write for tag authors. But it technically, works and could be refactored later. So I won't block, but please find somebody else to approve.

**Comment by jklymak** [comment]
Problems with examples should be issues for discussion, not tags.

**Comment by jklymak** [comment]
Because tags are for users to find things in the docs.  Issues are for maintainers to track issues.  

Basically by tagging an example as "too much" you are issuing a judgement.  To remove your judgement I would then have to open a PR to change the tag. That is backwards to how an issue works, where there are no committed changes until a PR is approved.

**Comment by timhoffm** [comment]
Is it sufficient to create a collection issue for "example rework"? Instead of adding a `internal: needs review` tag to the code, add a line in the issue

- [ ] internal: needs review: https://matplotlib.org/devdocs/gallery/statistics/errorbar.html

This is much more lightweight. It can be edited without a PR and review. Also it's possible to add a few words for context to the point. The tag must stand for itself.

The only downside I see is that non-core devs likely don't have the right to edit that issue. But that's not too bad. They can comment on it and we merge the comments in to one top-level list if needed.

**Comment by jklymak** [comment]
I'll stand by my stance documentation tags should not be used to organize issues.  There are _lots_ of mechanisms to organize and surface issues on GitHub, specifically tags and projects, and we should use those.

File: galleries/examples/statistics/confidence_ellipse.py
Comment (timhoffm) [inline galleries/examples/statistics/confidence_ellipse.py:223]: We have discussed this somewhere, but can't find the PR right now: I believe we wanted to use a standard formatting for better readability, and this can be
- one line for short lists, e.g.
  ```
  .. tags:: plot-type: speciality, domain: statistics
  ```
- one tag per line for longer lists
  ```suggestion
  #    plot-type: speciality
  #    plot-type: scatter
  #    component: ellipse
  #    component: patch
  #    domain: statistics
  ```

Comment (timhoffm) [reply galleries/examples/statistics/confidence_ellipse.py:223]: My definition of "long" is "does not fit on a single line". What's your cut-off? 😄

Comment (timhoffm) [reply galleries/examples/statistics/confidence_ellipse.py]: But you started a second line with `#    domain: statistics,`. To extrapolate, I could add more tags there, and when that's full, start a third line and continue.

Comment (timhoffm) [reply galleries/examples/statistics/confidence_ellipse.py]: No problem 😁

**Review by timhoffm — APPROVED** [review]

File: galleries/examples/statistics/multiple_histograms_side_by_side.py
Comment (timhoffm) [inline galleries/examples/statistics/multiple_histograms_side_by_side.py:67]: There's a warning in this file

> /home/circleci/project/doc/gallery/statistics/multiple_histograms_side_by_side.rst:101: WARNING: 'any' reference target not found: ../../_tags/domain-statistics-plot-type-barh-plot-type-histogram-styling-position [ref.any]

I suspect this is due to formatting, but not really sure

```suggestion
# .. tags::
#
#    domain: statistics
```

**Pull request merged**
